### PR TITLE
Fixes Traitor Uplinks Being Set to Existing Radio Channels

### LINKED ISSUE #56526: Radio Uplinks can spawn with common as their code, making them unusable
## Reproduction:
Radio uplinks can spawn with 145.9 as their code

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
Fixes #56526 
creates a new proc that returns a random unused radio frequency, tested to confirm that it does not return a used frequency

## Changelog
:cl:
fix: radio uplinks can no longer become unusable due to the code frequency
/:cl:

### CODE REVIEW

**Comment by Sparkezel** [comment]
Honestly all you needed to do in that case was to set radio uplink to any other channel, then return to common

**Comment by Kylerace** [comment]
oops forgot the message admins

File: code/__HELPERS/radio.dm [hunk 1]
```diff
@@ -8,12 +8,26 @@
...
 	return frequency * 10
+
+///returns a random unused frequency between MIN_FREE_FREQ & MAX_FREE_FREQ if free = TRUE, and MIN_FREQ & MAX_FREQ if FALSE
+/proc/return_unused_frequency(free = FALSE)
+	var/start = free ? MIN_FREE_FREQ : MIN_FREQ
+	var/end = free ? MAX_FREE_FREQ : MAX_FREQ
+
+	var/freq_to_check = rand(start, end)
+	freq_to_check += !(freq_to_check % 2) //make sure freq_to_check is odd
+	while(TRUE)
```
Comment (Jared-Fogle) [inline code/__HELPERS/radio.dm:28]: Change this to `while (!("[freq_to_check]" in GLOB.reverseradiochannels)), refactor with that.

File: code/__HELPERS/radio.dm [hunk 2]
```diff
@@ -8,12 +8,26 @@
...
 	frequency = text2num(frequency)
 	return frequency * 10
+
+///returns a random unused frequency between MIN_FREE_FREQ & MAX_FREE_FREQ if free = TRUE, and MIN_FREQ & MAX_FREQ if FALSE
+/proc/return_unused_frequency(free = FALSE)
+	var/start = free ? MIN_FREE_FREQ : MIN_FREQ
+	var/end = free ? MAX_FREE_FREQ : MAX_FREQ
+
+	var/freq_to_check = rand(start, end)
+	freq_to_check += !(freq_to_check % 2) //make sure freq_to_check is odd
```
Comment (Jared-Fogle) [inline code/__HELPERS/radio.dm:27]: This isn't readable, and it abuses that booleans aren't real. Just make an if statement.

File: code/__HELPERS/radio.dm (hunk 1 above)
Comment (Kylerace) [reply code/__HELPERS/radio.dm]: done

File: code/__HELPERS/radio.dm (hunk 2 above)
Comment (Kylerace) [reply code/__HELPERS/radio.dm]: done

**Pull request merged**